crates/matrix-sdk/src/event_cache/redecryptor.rs

```diff
@@ -256,10 +256,15 @@ impl EventCache {
 
                 // TODO: `replace_event_at()` propagates changes to the store for every event,
                 // we should probably have a bulk version of this?
-                state.replace_event_at(location, target_event).await?
+                state.replace_event_at(location, target_event.clone()).await?;
+                new_events.push(target_event);
             }
         }
 
+        // TODO: Is this useful? Do I need to call this once per event as well because
+        // replace event at works on a single event?
+        state.post_process_new_events(new_events, false).await?;
```

Comment:
This is the correct usage. Calling it with multiple events is better performance-wise.